tests/e2e/specs/in-game/one-offs/4_fours.spec.js

```diff
@@ -266,7 +258,7 @@ describe('FOURS', () => {
         p1Hand: [Card.TEN_OF_HEARTS, Card.SIX_OF_DIAMONDS],
         p1Points: [],
         p1FaceCards: [],
-        scrap: [Card.FOUR_OF_CLUBS, Card.ACE_OF_DIAMONDS, Card.FOUR_OF_SPADES, Card.FOUR_OF_DIAMONDS],
+        scrap: [Card.FOUR_OF_CLUBS, Card.ACE_OF_DIAMONDS, Card.FOUR_OF_SPADES],
```

Comment:
Okay so this is the crux of the issue with the failing test, and as you've said, it's a byproduct of a legitimate difference between the intended behavior of the GameState API vs legacy: specifically whether the 4 being played for a one-off hits the scrap when the targeted player declines to counter, or when the targeted player discards their cards.

The legacy approach put the oneOff in the scrap as soon as the targeted player declined to counter, and this looks to be the only assertion in the test suite that makes this behavior explicit. I think it actually doesn't matter much which way the system works, but it is more consistent that the oneOff go to the scrap when its effect is resolved since this is a requirement for threes (so you can't play a three and pick it back up in one move).

All that said, I think the most reasonable course of action is to simply remove this assertion from the test by taking the `scrap` out of the asserted fixture. Then it's ignored in this step and we can still assert that the 4 hits the scrap in the next assertGameState() call, which we do after the player discards.
```suggestion
```